### PR TITLE
Closes #148

### LINKED ISSUE #148: Create dataset loader for BioNLP-ST 2013 PC
- **Name:** BioNLP-ST 2013 PC
- **Description:** Event extraction data set for pathway curation
- **Task:** Event extraction
- **Paper:** https://aclanthology.org/W13-2009.pdf
- **Data:** [https://github.com/openbiocorpora/bionlp-st-2013-pc]()
- **License:** License Terms for BioNLP Shared Task 2013 Data

### PR DESCRIPTION
**Pull request by barthfab**
Please name your PR after the issue it closes. You can use the following line: "Closes #ISSUE-NUMBER" where you replace the ISSUE-NUMBER with the one corresponding to your dataset.

If the following information is NOT present in the issue, please populate:

- Name: BioNLP-ST 2013 PC
- Description: Event extraction data set for pathway curation
- Task: Event extraction
- Paper: https://aclanthology.org/W13-2009.pdf
- Data: [https://github.com/openbiocorpora/bionlp-st-2013-pc](https://github.com/bigscience-workshop/biomedical/issues/148)
- License: License Terms for BioNLP Shared Task 2013 Data

### Checkbox

- [x] Confirm that this PR is linked to the dataset issue.
- [x] Create the dataloader script `biodatasets/my_dataset/my_dataset.py` (please use only lowercase and underscore for dataset naming).
- [x] Provide values for the `_CITATION`, `_DATASETNAME`, `_DESCRIPTION`, `_HOMEPAGE`, `_LICENSE`, `_URLs`, `_SUPPORTED_TASKS`, `_SOURCE_VERSION`, and `_BIGBIO_VERSION` variables.
- [x] Implement `_info()`, `_split_generators()` and `_generate_examples()` in dataloader script.
- [x] Make sure that the `BUILDER_CONFIGS` class attribute is a list with at least one `BigBioConfig` for the source schema and one for a bigbio schema.
- [x] Confirm dataloader script works with `datasets.load_dataset` function.
- [x] Confirm that your dataloader script passes the test suite run with `python -m tests.test_bigbio biodatasets/my_dataset/my_dataset.py`.

### CODE REVIEW

File: biodatasets/bionlp_st_2013_pc/bionlp_st_2013_pc.py [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+from typing import List
+
+import datasets
+
+from utils import parsing, schemas
+from utils.configs import BigBioConfig
+from utils.constants import Tasks
+
+_DATASETNAME = "bionlp_st_2013_pc"
+_UNIFIED_VIEW_NAME = "bigscience"
```
Comment (sunnnymskang) [inline biodatasets/bionlp_st_2013_pc/bionlp_st_2013_pc.py:26]:
```suggestion
_UNIFIED_VIEW_NAME = "bigbio"
```
_UNIFIED_VIEW_NAME should match up the name of the existing config -either source or bigbio: see example/mlee.py

File: biodatasets/bionlp_st_2013_pc/bionlp_st_2013_pc.py [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+
+    SOURCE_VERSION = datasets.Version(_SOURCE_VERSION)
+    BIGBIO_VERSION = datasets.Version(_BIGBIO_VERSION)
+
+    BUILDER_CONFIGS = [
+        BigBioConfig(
+            name="bionlp_st_2013_pc_source",
+            version=SOURCE_VERSION,
+            description=_DESCRIPTION,
+            schema="source",
```
Comment (sunnnymskang) [inline biodatasets/bionlp_st_2013_pc/bionlp_st_2013_pc.py:84]: add subset_id for source config as well (subset_id = "bionlp_st_2013_pc")

File: biodatasets/bionlp_st_2013_pc/bionlp_st_2013_pc.py [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+    """the Pathway Curation (PC) task is a main event extraction task of the BioNLP shared task (ST) 2013."""
+
+    SOURCE_VERSION = datasets.Version(_SOURCE_VERSION)
+    BIGBIO_VERSION = datasets.Version(_BIGBIO_VERSION)
+
+    BUILDER_CONFIGS = [
+        BigBioConfig(
+            name="bionlp_st_2013_pc_source",
+            version=SOURCE_VERSION,
+            description=_DESCRIPTION,
```
Comment (sunnnymskang) [inline biodatasets/bionlp_st_2013_pc/bionlp_st_2013_pc.py:83]:
```suggestion
            description="bionlp_st_2013 source schema",
```

File: biodatasets/bionlp_st_2013_pc/bionlp_st_2013_pc.py [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+            schema="source",
+        ),
+        BigBioConfig(
+            name="bionlp_st_2013_pc_bigbio_kb",
+            version=BIGBIO_VERSION,
+            description=_DESCRIPTION,
```
Comment (sunnnymskang) [inline biodatasets/bionlp_st_2013_pc/bionlp_st_2013_pc.py:89]:
```suggestion
            description="bionlp_st_2013_pc BigBio schema",
```

File: biodatasets/bionlp_st_2013_pc/bionlp_st_2013_pc.py [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+            schema="bigbio_kb",
+            subset_id="bionlp_st_2013_pc",
+        ),
+    ]
+
+    DEFAULT_CONFIG_NAME = _DATASETNAME
```
Comment (sunnnymskang) [inline biodatasets/bionlp_st_2013_pc/bionlp_st_2013_pc.py:95]: DEFAULT_CONFIG_NAME should be either of the name of the config defined as above (either "bionlp_st_2013_pc_source" or "bionlp_st_2013_pc_bigbio_kb"

**Review by sunnnymskang — CHANGES_REQUESTED** [review]

File: biodatasets/bionlp_st_2013_pc/bionlp_st_2013_pc.py (hunk 3 above)
Comment (hakunanatasha) [reply biodatasets/bionlp_st_2013_pc/bionlp_st_2013_pc.py]: fixed

File: biodatasets/bionlp_st_2013_pc/bionlp_st_2013_pc.py (hunk 4 above)
Comment (hakunanatasha) [reply biodatasets/bionlp_st_2013_pc/bionlp_st_2013_pc.py]: fixed

File: biodatasets/bionlp_st_2013_pc/bionlp_st_2013_pc.py (hunk 5 above)
Comment (hakunanatasha) [reply biodatasets/bionlp_st_2013_pc/bionlp_st_2013_pc.py]: I've seen the defaults usually go to source.

File: biodatasets/bionlp_st_2013_pc/bionlp_st_2013_pc.py [hunk 6]
```diff
@@ -0,0 +1,235 @@
...
+
+_HOMEPAGE = "https://github.com/openbiocorpora/bionlp-st-2013-pc"
+
+_LICENSE = "https://creativecommons.org/licenses/by/3.0/ CC BY-SA 3.0"
+
+_URLs = {
+    "bionlp_st_2013_pc": "https://github.com/openbiocorpora/bionlp-st-2013-pc/archive/refs/heads/master.zip",
+}
+
+_SUPPORTED_TASKS = [Tasks.EVENT_EXTRACTION]
```
Comment (leonweber) [inline biodatasets/bionlp_st_2013_pc/bionlp_st_2013_pc.py]: This should also include NER and COREF

**Review by leonweber — APPROVED** [review]
Looks good to me. Failure for test_schema is expected because in all BioNLP shared task datasets test set annotations are hidden and thus missing from our version.

**Review by leonweber — APPROVED** [review]
Looks good to me. Failure for test_schema is expected because in all BioNLP shared task datasets test set annotations are hidden and thus missing from our version.

**Review by leonweber — APPROVED** [review]

**Pull request merged**